notebooks/cuml_benchmarks.ipynb

```diff
@@ -0,0 +1,997 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Benchmark and Bounds Tests\n",
+    "\n",
+    "The purpose of this notebook is to benchmark all of the single GPU cuML algorithms against their skLearn counterparts, while also providing the ability to find and verify upper bounds.\n",
+    "\n",
+    "Each benchmark returns a Panda with the results, which can then be analyzed, manipulated, and stored to disk. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Notebook Credits\n",
+    "**Authorship**\n",
+    "Original Author: Corey Nolet <br />\n",
+    "Last Edit: Taurean Dyer, 9/25/2019<br />\n",
+    "\n",
+    "Last Edit: Corey Nolet, 10/04/2019\n",
+    "    \n",
+    "### Test System Specs\n",
+    "Test System Hardware: DGX-1 \n",
+    "Test System Software: Ubuntu 16.04  \n",
+    "RAPIDS Version: 0.10.0pre - Conda Install  \n",
+    "Driver: 410.48\n",
+    "CUDA: 10.0  \n",
+    "\n",
+    "### Known Working Systems\n",
+    "RAPIDS Versions: 0.10+"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "0.14.0a+852.gc5770a0.dirty\n"
+     ]
+    }
+   ],
+   "source": [
+    "import cuml\n",
+    "import pandas as pd\n",
+    "\n",
+    "from cuml.benchmark.runners import SpeedupComparisonRunner\n",
+    "from cuml.benchmark.algorithms import algorithm_by_name\n",
+    "\n",
+    "print(cuml.__version__)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "N_REPS = 3  # Number of times each test is repeated\n",
+    "\n",
+    "DATA_NEIGHBORHOODS = \"blobs\"\n",
+    "DATA_CLASSIFICATION = \"classification\"\n",
+    "DATA_REGRESSION = \"regression\"\n",
+    "DATA_NLP = \"20newsgroups\"\n",
+    "\n",
+    "INPUT_TYPE = \"numpy\"\n",
+    "\n",
+    "benchmark_results = []\n",
+    "\n",
+    "def enrich_result(algorithm, runner, result):\n",
+    "    result[\"algo\"] = algorithm\n",
+    "    result[\"dataset_name\"] = runner.dataset_name\n",
+    "    result[\"input_type\"] = runner.input_type\n",
+    "    return result\n",
+    "\n",
+    "def execute_benchmark(algorithm, runner):\n",
+    "    results = runner.run(algorithm_by_name(algorithm), verbose=True)\n",
+    "    results = [enrich_result(algorithm, runner, result) for result in results]\n",
+    "    benchmark_results.extend(results)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
```

Comment:
What do you think about having just two ranges of row sizes, like in pybench? A small range (16k, 32k, 64k) and a large range (250k, 1M, 4M?)? Then we can specify them at the top and reuse them everywhere (small for slow algos). That would make it easier to play around with ranges and e.g. shrink them when debugging